pkg/readiness/pruner/pruner_test.go

```diff
@@ -184,3 +195,30 @@ func Test_ExpectationsMgr_DeletedSyncSets(t *testing.T) {
 		})
 	}
 }
+
+// Test_ExpectationsMgr_missedInformers verifies that the pruner can handle a scenario
+// where the readiness tracker's state will never match the informer cache events.
+func Test_ExpectationsMgr_missedInformers(t *testing.T) {
+	ctx, cancelFunc := context.WithCancel(context.Background())
+
+	// because we will use a separate lister for the tracker from the mgr client
+	// the contents of the readiness tracker will be superset of the contents of the mgr's client
+	tl := fakes.NewTestLister(
+		fakes.WithSyncSets([]*syncsetv1alpha1.SyncSet{
+			fakes.SyncSetFor("syncset-1", []schema.GroupVersionKind{podGVK, configMapGVK}),
+		}),
+	)
+	em, _ := setupTest(ctx, t, testOptions{testLister: tl})
+
+	require.Eventually(t, func() bool {
+		return em.tracker.SyncSourcesSatisfied()
+	}, timeout, tick, "waiting on sync sources to get satisfied")
+
+	em.pruneNotWatchedGVKs()
```

Comment:
Considering the ExpectationsPruner is a runnable and is hooked into the manager IRL, we'd have a more realistic test if we ran the EP in the manager.  This test is already creating a manager, so it's just a question of plumbing the right stuff.

Reply:
I don't mind doing this for this test but I think I had gotten some feedback about  making direct calls in testing scenarios like these from @maxsmythe  (please correct me if I'm wrong).

Reply:
Avoiding multi-threading in unit tests (where possible) helps to avoid flakiness. If we wanted to test that something was added to the manager and started (not sure we do, there's not a lot that can go wrong there), it would probably belong in a different test. Hopefully indirectly tested via e2e tests.